WalletWasabi/Wallets/PasswordFinder.cs

```diff
@@ -0,0 +1,68 @@
+using NBitcoin;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security;
+
+namespace WalletWasabi.Wallets
+{
+	public static class PasswordFinder
+	{
+		public static readonly Dictionary<string, string> Charsets = new Dictionary<string, string>
+		{
+			["en"] = "abcdefghijkmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ",
+			["es"] = "aábcdeéfghiíjkmnñoópqrstuúüvwxyzAÁBCDEÉFGHIÍJKLMNNOÓPQRSTUÚÜVWXYZ",
+			["pt"] = "aáàâābcçdeéêfghiíjkmnoóôōpqrstuúvwxyzAÁÀÂĀBCÇDEÉÊFGHIÍJKMNOÓÔŌPQRSTUÚVWXYZ",
+			["it"] = "abcdefghimnopqrstuvxyzABCDEFGHILMNOPQRSTUVXYZ",
+			["fr"] = "aâàbcçdæeéèëœfghiîïjkmnoôpqrstuùüvwxyÿzAÂÀBCÇDÆEÉÈËŒFGHIÎÏJKMNOÔPQRSTUÙÜVWXYŸZ",
+		};
+
+		public static string? Find(Wallet wallet, string language, bool useNumbers, bool useSymbols, string likelyPassword, Action<int> reportPercentage)
```

Comment:
I would prefer a component-like design where most of these parameters are in the constructor because or properties. Also, instead of a delagate it could be possible to have an event.

Reply:
Generally, I agree with your concept. I just started to think about what would I put into the constructor when I am using this in a viewmodel. Basically the user will set the parameters and press the find button. PW finder has no lifecycle. I can make it as a class but it will be a one liner in both cases.